### PR TITLE
fix(react-editor): reset focus offset when triple clicking

### PR DESCRIPTION
**Pull request by bytrangle**
**Description**

In my triple click fix, I reset the `focusNode` but forgot to reset the related `focusOffset` value. As a result, the Slate range computed from DOM selection is still not correct and cause a number of issues when users try to transform a block by applying formatting or deleting. 

**Issue**
Fixes: #4492 

**Example**
Before the fix

https://user-images.githubusercontent.com/28924121/133071690-27d21192-b86c-4b6a-9741-0f2f1ff53218.mp4

After the fix

- Selecting a block by clicking and dragging, then hitting Backspace should remove that block.

https://user-images.githubusercontent.com/28924121/133098079-78fdccd1-3eef-42a3-b852-ca0ba55453d9.mp4

- Selecting a block through clicking and hitting Shift + Arrow down, then hitting Backspace should remove that block.

https://user-images.githubusercontent.com/28924121/133098140-5d2963af-0014-4501-a117-b4b6d71f1d91.mp4

- Triple click at the beginning of a Mark should delete the whole block that contains that Mark.

https://user-images.githubusercontent.com/28924121/133098182-1198893c-aa2c-4291-b777-ab5c44f06643.mp4

**Context**
Here's what I did:
- Refactor the condition for checking for triple clicking or Shift + down arrow. Chrome seems to have issue with selection that is not clicking and dragging.
- Rewrite the code for getting `focusNode` to avoid error thrown when accessing property of a potentially null value.
- Add logic for getting the DOM Text node of the `focusNode` object.

**Checks**
- [x] The new code matches the existing patterns and styles.
- [x] The tests pass with `yarn test`.
- [x] The linter passes with `yarn lint`. (Fix errors with `yarn fix`.)
- [x] The relevant examples still work. (Run examples with `yarn start`.)
- [ ] You've [added a changeset](https://github.com/atlassian/changesets/blob/master/docs/adding-a-changeset.md) if changing functionality. (Add one with `yarn changeset add`.)

### CODE REVIEW

File: packages/slate-react/src/plugin/react-editor.ts [hunk 1]
```diff
@@ -572,36 +572,52 @@ export const ReactEditor = {
...
-            // Get the element node that holds the focus text node
-            // Not every Slate element node contains a child node with `data-slate-string`,
-            // such as void nodes, so the result below could be null.
-            const innermostFocusElement = focusElement!.querySelector(
-              '[data-slate-string]'
-            )
-            if (innermostFocusElement) {
-              const lastTextNode = innermostFocusElement.childNodes[0]
-              focusNode = lastTextNode
+          // if (anchorNode!.nodeType === 1) {
```
Comment (dylans) [inline packages/slate-react/src/plugin/react-editor.ts:584]: Just remove the commented out code.

**Review by dylans — CHANGES_REQUESTED** [review]

File: packages/slate-react/src/plugin/react-editor.ts (hunk 1 above)
Comment (bytrangle) [reply packages/slate-react/src/plugin/react-editor.ts:584]: I'm not sure what you meant. Which part is the comment-out code?

**Review by dylans — APPROVED** [review]

**Comment by dylans** [comment]
Sigh, I merged without a changeset. opening a PR for the changeset now.

**Comment by bytrangle** [comment]
I forgot to ask a noob question: When should I add a changeset, and when should I not? I thought this change is a correction for my flawed triple click fix, so I didn't think of adding a changeset.

**Pull request merged**